### PR TITLE
Add pinocchio recipe

### PR DESCRIPTION
**Pull request by lesteve**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by lesteve** [comment]
Glad to see there are not more unexpected cmake issues showing up. I guess the next steps are:
* get the eigenpy #8820 PR merged.
* do PRs for the other dependencies like `urdfdom` and `console_bridge`.

update the `pinocchio` recipe once this is done.

**Comment by jcarpent** [comment]
I confirm that I want to be included in the list of maintainers. Thanks @lesteve for all this nice work.

File: recipes/pinocchio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (jan-janssen) [inline recipes/pinocchio/meta.yaml]: Please remove the comments in line 1-5 and 14-15

Comment (lesteve) [reply recipes/pinocchio/meta.yaml]: Thanks for the feed-back and sorry I forgot about the comments.

Note I marked the PR WIP as an attempt to indicate I was not sollicitating reviews for now. I edited the top post to make this more clear.

Comment (lesteve) [reply recipes/pinocchio/meta.yaml]: Of course @jan-janssen if you have a better way make the WIP status of this PR more imediate this for potential reviewers, I'd be happy to hear about it!

Note that it is extremely useful to have the CIs (this definitely surfaced a few problems/limitations in the pinocchio cmake files) so the draft PR github functionality (from what I have heard CI does not run on a draft PR) ...

**Comment by lesteve** [comment]
All the dependencies related to URDF support are in conda-forge now.

@conda-forge/staged-recipes I think this is ready for review.

@wolfv can you add a comment saying you are fine being a maintainer for `pinocchio`?

File: recipes/pinocchio/build.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+mkdir build
+cd build
+
+cmake .. \
+      -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=$PREFIX -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib
+
+make
```
Comment (chrisburr) [inline recipes/pinocchio/build.sh]: Can multiple cores be used?
```suggestion
make -j${CPU_COUNT}
```

File: recipes/pinocchio/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+{% set version = "2.1.5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # url: https://github.com/stack-of-tasks/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  # sha256: 773c0cb037c1fbddda4668dd357d65cd6914566462b4faf5f37295964e35c996
+  git_url: https://github.com/stack-of-tasks/{{ name }}.git
```
[the remark this answers is not in the document]
Comment (lesteve) [reply recipes/pinocchio/meta.yaml]: Sure I forgot about this sorry ...

File: recipes/pinocchio/build.sh (hunk 2 above)
Comment (lesteve) [reply recipes/pinocchio/build.sh]: Out of curiosity how many cores are available on Azure Pipelines?

File: recipes/pinocchio/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lesteve) [reply recipes/pinocchio/meta.yaml]: Actually @jcarpent I am pretty sure the `devel` branch has some fixes for the cmake files. Would it be possible to cut a new `pinocchio` release for these fixes?

File: recipes/pinocchio/build.sh (hunk 2 above)
Comment (chrisburr) [reply recipes/pinocchio/build.sh]: I can't remember exactly but it's either 4 or 8 and I think macOS has more than Linux.

**Comment by lesteve** [comment]
A new pinocchio release has been made a few hours ago. The recipe has been updated to use the latest release and the CI is now green.

@conda-forge/staged-recipes reviews more than welcome!

**Comment by lesteve** [comment]
Chatting with @jcarpent, we may put this PR on hold, because we'd like to add additional dependencies in conda-forge. Putting this PR back to a WIP state!

**Comment by wolfv** [comment]
what other deps? any from the ROS ecosystem?

**Comment by lesteve** [comment]
`pinocchio` will be more useful with [hpp-fcl](https://github.com/humanoid-path-planner/hpp-fcl). `hpp-fcl` depends on:
- `assimp` already in `conda-forge`, thanks a lot @wolfv!
- `octomap` conda-forge/staged-recipes at https://github.com/conda-forge/staged-recipes/pull/9056

Once `hpp-fcl` is in conda-forge I think we are good to go with the `pinocchio` recipe.

**Comment by wolfv** [comment]
cool, yeah, happy that this was useful! I need `fcl` as well (not `hpp-fcl` unfortunately, though).

**Comment by lesteve** [comment]
`fcl` needs `octomap` so you'll benefit from #9056.

**Comment by lesteve** [comment]
OK, we now have added all the `pinocchio` dependencies we cared about in conda-forge.

@conda-forge/staged-recipes this PR is ready to review!

@jcarpent aggreed to be a maintainer in https://github.com/conda-forge/staged-recipes/pull/8844#issuecomment-513659438.

@wolfv if you can post a comment saying that you agree to be a maintainer, that'd be great!

**Comment by wolfv** [comment]
That's great! I agree, of course

**Pull request merged**